applications/tari_merge_mining_proxy/src/proxy.rs

```diff
@@ -605,20 +642,43 @@ impl InnerService {
             }
         }
 
-        for monerod_url in self.config.monerod_url.iter() {
-            let uri = format!("{}{}", monerod_url, uri.path()).parse::<Url>()?;
-            match reqwest::get(uri.clone()).await {
-                Ok(_) => {
-                    let mut lock = self.last_available_server.write().expect("Write lock should not fail");
-                    *lock = Some(monerod_url.to_string());
-                    info!(target: LOG_TARGET, "Monerod server available: {:?}", uri.clone());
-                    return Ok(uri);
-                },
-                Err(_) => {
-                    warn!(target: LOG_TARGET, "Monerod server unavailable: {:?}", uri);
-                    continue;
-                },
+        let current_url = {
```

Comment:
This will always be `"".to_string()` because of the above return statement, which makes the extra code here and the find below unnecessary

Reply:
Good pickup, I will fix the logic error here. If we have a troublesome `monerod` we do not want to connect to it straight away again even if it is possible; much better to try the next server in the list. `self.last_available_server` is zeroed if `async fn proxy_request_to_monerod` returns an error.

Reply:
Done